Bug [bryan_snaily]: Problem with communication with SnailyCAD API

Player tried to create a character, with both qb-multicharacter and Renzu-multicharacter but script return the below error.

### FX Server logs
```properties
[ script:bryan_snaily] SCRIPT ERROR: @bryan_snaily/server/server.lua:317: [Error] This API seems to be outdated, please update the script to the newest release
[ script:bryan_snaily] > userCallback (@bryan_snaily/server/server.lua:317)
```

### SnailyCAD logs
```js
[2024-02-27T18:08:31.072] [ERROR] [SnailyCAD] - {
  method: 'GET',
  url: '/v1admin/values/gender',
  route: '/v1admin/values/gender',
  headers: {
    connection: 'upgrade',
    host: '3.5-api.legacyrp.org',
    'x-forwarded-for': '192.168.10.1',
    'content-length': '2',
    'user-agent': 'FXServer/PerformHttpRequest',
    accept: '*/*',
    'content-type': 'application/json',
    'snaily-cad-api-token': 'sng_wvFCwSONL9lno5YrfVgdL7P1bc7gbRV39ZZzsDRQOizD-dZe6rlE_vE-'
  },
  body: [],
  query: {},
  params: {},
  reqId: 'e459a162a345496a917e89174cba94e4',
  time: 2024-02-27T23:08:31.072Z,
  duration: 6,
  event: 'request.end',
  status: 404,
  status_code: '404',
  state: 'KO'
}
```
### Used Versions
SnailyCad: latest
FX Server: 7538
bryan_snaily script: V1.0.3
Renzu-multicharacter: latest

Original thread: [Discord](https://discord.com/channels/1188124299144466492/1210696909245780148)

--- a/Bryan-Snaily-CAD-Integration-main/bryan_snaily/fxmanifest.lua
+++ b/Bryan-Snaily-CAD-Integration-main/bryan_snaily/fxmanifest.lua
@@ -1,9 +1,14 @@
 fx_version 'cerulean'
+
 game 'gta5'
 
 author 'BryaN'
 
-shared_script 'config.lua'
+shared_scripts {
+    'config.lua',
+    'shared/*.lua'
+}
+
 server_scripts {
     'server/*.lua'
 }

--- a/Bryan-Snaily-CAD-Integration-main/bryan_snaily/server/server.lua
+++ b/Bryan-Snaily-CAD-Integration-main/bryan_snaily/server/server.lua
@@ -29,7 +29,7 @@ InsertNewCitizen = function(firstname, lastname, birthdate, gender, ethnicity, h
         return
     end
 
-    PerformHttpRequest(string.format('%sadmin/import/citizens', Config.API.URL),
+    PerformHttpRequest(string.format('%sadmin/import/citizens', URLNormalize(Config.API.URL)),
         function(errorCode, resultData, resultHeaders, errorData)
             if (errorCode ~= 200) then
                 if errorCode == 404 then
@@ -69,7 +69,7 @@ InsertNewWeapon = function(weaponHash, firstname, lastname)
         return
     end
 
-    PerformHttpRequest(string.format('%sadmin/import/weapons', Config.API.URL),
+    PerformHttpRequest(string.format('%sadmin/import/weapons', URLNormalize(Config.API.URL)),
         function(errorCode, resultData, resultHeaders, errorData)
             if (errorCode ~= 200) then
                 if errorCode == 404 then
@@ -114,7 +114,7 @@ InsertNewVehicle = function(plate, vehicleHash, firstname, lastname, color)
         return
     end
 
-    PerformHttpRequest(string.format('%sadmin/import/vehicles', Config.API.URL),
+    PerformHttpRequest(string.format('%sadmin/import/vehicles', URLNormalize(Config.API.URL)),
         function(errorCode, resultData, resultHeaders, errorData)
             if (errorCode ~= 200) then
                 if errorCode == 404 then
@@ -158,7 +158,7 @@ SetLicenseToCitizen = function(license, value, firstname, lastname)
         return
     end
 
-    PerformHttpRequest(string.format('%ssearch/actions/licenses/%s', Config.API.URL, citizenId),
+    PerformHttpRequest(string.format('%ssearch/actions/licenses/%s', URLNormalize(Config.API.URL), citizenId),
         function(errorCode, resultData, resultHeaders, errorData)
             if (errorCode ~= 200) then
                 if errorCode == 404 then
@@ -310,7 +310,7 @@ GetValueId = function(table, value, columnToCheck)
     local awaitingResponse = true
     local valueId
 
-    PerformHttpRequest(string.format('%sadmin/values/%s', Config.API.URL, table),
+    PerformHttpRequest(string.format('%sadmin/values/%s', URLNormalize(Config.API.URL), table),
         function(errorCode, resultData, resultHeaders, errorData)
             if (errorCode ~= 200) then
                 if errorCode == 404 then
@@ -360,7 +360,8 @@ GetCitizenId = function(firstname, lastname)
         error('[Error] Lastname is required', 2)
     end
 
-    PerformHttpRequest(string.format('%sadmin/manage/citizens?query=%s+%s', Config.API.URL, firstname, lastname),
+    PerformHttpRequest(
+        string.format('%sadmin/manage/citizens?query=%s+%s', URLNormalize(Config.API.URL), firstname, lastname),
         function(errorCode, resultData, resultHeaders, errorData)
             if (errorCode ~= 200) then
                 if errorCode == 404 then

--- a/Bryan-Snaily-CAD-Integration-main/bryan_snaily/shared/url_normalize.lua
+++ b/Bryan-Snaily-CAD-Integration-main/bryan_snaily/shared/url_normalize.lua
@@ -1,0 +1,9 @@
+---Normalize the given url by appending slash to the end of the url
+---@param url string
+---@return string
+function URLNormalize(url)
+  if string.match(url, '/$') == nil then
+    return url .. '/'
+  end
+  return url
+end

